test/run-tests

```diff
@@ -370,18 +387,47 @@ def choose_task(gh, repos, images):
             for image in images:
                 status = statuses.get("linux-system-roles-test/" + image["name"])
 
-                # Add a task if there's no status for it yet or the status is
-                # pending without a hostname in the description
-                if not status or (
-                    status["state"] == "pending" and not status.get("description")
-                ):
+                if check_commit_needs_testing(gh, status, owner, repo, number):
                     task = Task(owner, repo, number, head, image)
                     tasks.append(task)
 
     if tasks:
         return random.choice(tasks)
 
 
+def check_commit_needs_testing(gh, status, owner, repo, pullnr):
+    """ Check if commit needs to be checked """
+    # Check it, if there's no status for it yet
+    if not status:
+        return True
+
+    # or the status is pending without a hostname in the description
+    if status["state"] == "pending" and not status.get("description"):
+        return True
+
+    last_status_update = dateutil.parser.parse(status["updated_at"])
+    comments = reversed(get_comments(gh, owner, repo, pullnr, after=last_status_update))
+
```

Comment:
What should this protect against? The commits are already whitelisted and they might only overload the CI system. If they see that there was an error, they can re-request a CI check, so we do not have to. But I do not mind restricting this to members, too.

Reply:
Is there any likely case where a retest would work without a new commit and any change to the test infrastructure (which presumably only members can do)?

Reply:
Why do you think it is because of the connection to the VM is lost?
Yes, in the log there is `Shared connection to 127.0.0.3 closed.` but note this is always there, even for successful runs. I think the problem lies elsewhere. (And is related to the non-zero exit code.)
Anyway, if re-doing the test works, it is a valid point to allow it like you are doing. But we should fix the underlying problem eventually.

Reply:
> I think the problem lies elsewhere. (And is related to the non-zero exit code.)

I suggest to replace -q by -v in the dnf command line to debug that more easily

Reply:
you are right about the `closed` message, I only compared it to the last lines of successful runs, and there it does not appear https://larsu.fedorapeople.org/logs/linux-system-roles-network-pull-69-7c5dce7-fedora-28-20180711-104714/artifacts/ansible.log - but it seems to be related to the raw tasks. I changed it to use `-v` and removed `sudo`.